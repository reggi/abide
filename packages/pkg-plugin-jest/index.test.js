import pluginJest from './index'

test('pluginJest', () => {
  const result = pluginJest({pkgrc: [], pkg: {}, opt: {}})
  expect(result).toMatchSnapshot()
})

test('pluginJest: babel, hunderedPercent', () => {
  const result = pluginJest({pkgrc: [], pkg: {}, opt: {babel: true, hunderedPercent: true}})
  expect(result).toMatchSnapshot()
})

test('pluginJest: babel, hunderedPercent, forceCoverageMatch', () => {
  const result = pluginJest({
    pkgrc: [],
    pkg: {},
    opt: {
      babel: true,
      hunderedPercent: true,
      forceCoverageMatch: 'index.js'
    }
  })
  expect(result).toMatchSnapshot()
})

test('pluginJest: with overwrite', () => {
  const pkg = {
    'scripts': {
      'test': 'original example'
    }
  }
  const result = pluginJest({
    overwrite: true,
    pkgrc: [],
    pkg: pkg,
    opt: {
      babel: true,
      hunderedPercent: true,
      forceCoverageMatch: 'index.js'
    }
  })
  expect(result).toMatchSnapshot()
})

test('pluginJest: without overwrite', () => {
  const pkg = {
    'scripts': {
      'test': 'original example'
    }
  }
  const result = pluginJest({
    overwrite: false,
    pkgrc: [],
    pkg: pkg,
    opt: {
      babel: true,
      hunderedPercent: true,
      forceCoverageMatch: 'index.js'
    }
  })
  expect(result).toMatchSnapshot()
})
