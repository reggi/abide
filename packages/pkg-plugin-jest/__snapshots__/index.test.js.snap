// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pluginJest 1`] = `
Object {
  "devDependencies": Object {
    "jest": "^22.4.2",
  },
  "jest": Object {},
  "scripts": Object {
    "jest": "jest",
    "jest:coverage": "jest --coverage",
  },
}
`;

exports[`pluginJest: babel, hunderedPercent 1`] = `
Object {
  "devDependencies": Object {
    "babel-jest": "^22.4.1",
    "jest": "^22.4.2",
  },
  "jest": Object {
    "collectCoverage": true,
    "coverageThreshold": Object {
      "global": Object {
        "branches": 100,
        "functions": 100,
        "lines": 100,
        "statements": 100,
      },
    },
  },
  "scripts": Object {
    "jest": "jest",
    "jest:coverage": "jest --coverage",
  },
}
`;

exports[`pluginJest: babel, hunderedPercent, forceCoverageMatch 1`] = `
Object {
  "devDependencies": Object {
    "babel-jest": "^22.4.1",
    "jest": "^22.4.2",
  },
  "jest": Object {
    "collectCoverage": true,
    "coverageThreshold": Object {
      "global": Object {
        "branches": 100,
        "functions": 100,
        "lines": 100,
        "statements": 100,
      },
    },
    "forceCoverageMatch": Array [
      "index.js",
    ],
  },
  "scripts": Object {
    "jest": "jest",
    "jest:coverage": "jest --coverage",
  },
}
`;

exports[`pluginJest: with overwrite 1`] = `
Object {
  "devDependencies": Object {
    "babel-jest": "^22.4.1",
    "jest": "^22.4.2",
  },
  "jest": Object {
    "collectCoverage": true,
    "coverageThreshold": Object {
      "global": Object {
        "branches": 100,
        "functions": 100,
        "lines": 100,
        "statements": 100,
      },
    },
    "forceCoverageMatch": Array [
      "index.js",
    ],
  },
  "scripts": Object {
    "jest": "jest",
    "jest:coverage": "jest --coverage",
    "test": "original example",
  },
}
`;

exports[`pluginJest: without overwrite 1`] = `
Object {
  "devDependencies": Object {
    "babel-jest": "^22.4.1",
    "jest": "^22.4.2",
  },
  "jest": Object {
    "collectCoverage": true,
    "coverageThreshold": Object {
      "global": Object {
        "branches": 100,
        "functions": 100,
        "lines": 100,
        "statements": 100,
      },
    },
    "forceCoverageMatch": Array [
      "index.js",
    ],
  },
  "scripts": Object {
    "jest": "jest",
    "jest:coverage": "jest --coverage",
    "test": "original example",
  },
}
`;
